File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        
    int n = height.size();
    
    int maximumArea = INT_MIN;

    int i=0;
    int j=n-1;

    while(i<j){

        int width = (j-i);
        int heights = min(height[i], height[j]);

        int currArea = width*heights;

        maximumArea = max(maximumArea, currArea);

        if(height[i] < height[j]){
            i++;
        } 
        else{
            j--;
        }
    }

    return maximumArea;
    }
};

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        l, r = 0, len(height)-1
        res=0
        
        while l<r:
            
            width = r-l
            h = 0
            if height[l] < height[r]:
                h = height[l]
                l += 1
            else:
                h = height[r]
                r -= 1
            
            currArea = h * width
            res = max(currArea, res)
            
        return res
            